File: sherpout-server/src/main/java/com/sherpout/server/api/training/dto/TrainingPlanExerciseDTO.java
package com.sherpout.server.api.training.dto;

import com.sherpout.server.api.exercise.dto.ExerciseSelectDTO;
import jakarta.validation.Valid;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class TrainingPlanExerciseDTO {
    private Long id;

    @Valid
    @NotNull
    private ExerciseSelectDTO exercise;

    @NotNull
    @Min(1)
    @Max(16)
    private Integer sets;
}

File: sherpout_mobile/lib/common/dto/training_plan_exercise_dto.dart
import 'exercise_select_dto.dart';

class TrainingPlanExerciseDTO {
  int? id;
  ExerciseSelectDTO? exercise;
  int? sets;

  TrainingPlanExerciseDTO({
    this.id,
    this.exercise,
    this.sets,
  });

  factory TrainingPlanExerciseDTO.fromJson(Map<String, dynamic> json) {
    return TrainingPlanExerciseDTO(
      id: json['id'] != null ? json['id'] as int : null,
      exercise: ExerciseSelectDTO.fromJson(json['exercise']),
      sets: json['sets'] != null ? json['sets'] as int : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      if (exercise != null) 'exercise': exercise!.toJson(),
      if (sets != null) 'sets': sets,
    };
  }
}